File: Models/DonacionSolicitud.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace Donaciones.Models
{
    public class DonacionSolicitud:Base
    {
        public string Titulo { get; set; }
        public string Observacion { get; set; }

        public int CantidadActual => this.Donaciones.Sum(w => w.Cantidad);
        public int CantidadFin { get; set; }
        public int? OrganizacionId { get; set; }
        public Organizacion Organizacion { get; set; }
        public int? OwnerId { get; set; }
        public Usuario Owner { get; set; }
        public int EstadoId { get; set; }
        public Estado Estado { get; set; }

        public IEnumerable<Donacion> Donaciones { get; set; }
    }
}

File: ClientApp/src/app/models/donacionSolicitud.ts
import { IBase } from "./base";
import { IDonacion } from "./donacion";
import { IDonacionSolicitudItem } from "./donacionSolicitudItem";
import { IEstado } from "./estado";
import { IOrganizacion } from "./organizacion";
import { IUsuario } from "./usuario";

export interface IDonacionSolicitud extends IBase{
    titulo: string
    observacion: string
    cantidadActual:number
    cantidadFin:number
    organizacionId: number
    organizacion: IOrganizacion
    ownerId:number
    owner:IUsuario
    estadoId:number
    estado:IEstado
    donaciones: IDonacion[]
}
